Lib/test/test_pathlib.py

```diff
@@ -3002,6 +3002,25 @@ def test_absolute(self):
             self.assertEqual(str(P('a', 'b', 'c').absolute()),
                              os.path.join(share, 'a', 'b', 'c'))
 
+        drive = os.path.splitdrive(BASE)[0]
+        with os_helper.change_cwd(BASE):
+            # Relative path with root
+            self.assertEqual(str(P('\\').absolute()), drive + '\\')
+            self.assertEqual(str(P('\\foo').absolute()), drive + '\\foo')
+
+            # Relative path on current drive
+            self.assertEqual(str(P(drive).absolute()), BASE)
+            self.assertEqual(str(P(drive + 'foo').absolute()), os.path.join(BASE, 'foo'))
+
+        with os_helper.subst_drive(BASE) as other_drive:
+            other_cwd = f'{other_drive}\\dirA'
+            with os_helper.change_cwd(other_cwd):
+                pass
```

Comment:
I forgot the suggestion.

```suggestion
            # set the working directory on the substitute drive
            saved_dir = os.getcwd()
            os.chdir(other_cwd)
            os.chdir(saved_dir)
```